util/econftool.c

```diff
@@ -0,0 +1,612 @@
+ * @par group2  The 2nd group to compare.
+ * @par new     Elements that are not found in group 1 but in group 2 are stored here.
+ * @par deleted Elements that are found in group 1 but not in group 2 are stored here.
+ * @par common  Elements that are found in group 1 and group 2 are stored here.
+ * @par size_g1 The number of elements in group 1.
+ * @par size_g2 The number of elements in group 2.
+ * @par size_new The number of elements in new
+ * @par size_deleted The number of elements in deleted
+ * @par size_common The number of elements in common.
+ * @return void.
+ */
+static void diffGroups(char **group1, char **group2, char ***new, char ***deleted, char ***common,
+                       size_t *size_g1, size_t *size_g2, size_t *size_new, size_t *size_deleted,
+                       size_t *size_common)
+{
+    size_t size = *size_g1 + *size_g2 + 1;
+    bool found1 = false;
+    bool found2 = false;
+    *new = calloc(size, sizeof(char*));
+    *deleted = calloc(*size_g1 + 1, sizeof(char*));
+    *common = calloc(*size_g1 + 1, sizeof(char*));
+
+    if (*new == NULL || *deleted == NULL || *common == NULL) {
+        fprintf(stderr, "Error with calloc()!");
+        exit(EXIT_FAILURE);
+    }
+    /* compare group1 with group2 */
+    for(size_t i = 0; i < *size_g1; i++) {
+        found1 = false;
+        for (size_t j = 0; j < *size_g2; j++) {
+            if (strcmp(group1[i],group2[j]) == 0) {
+                (*common)[*size_common] = strdup(group1[i]);
+                (*size_common)++;
+                found1 = true;
+                break;
+            }
+        }
+        if (!found1) {
+            /* inside group1 but not in group2 -> deleted element */
+            (*deleted)[*size_deleted] = strdup(group1[i]);
+            (*size_deleted)++;
+        }
+    }
+    /* compare group2 with common */
+    for (size_t j = 0; j < *size_g2; j++) {
+        found2 = false;
+        for(size_t i = 0; i < *size_common; i++) {
+            if (strcmp(group2[j], (*common)[i]) == 0) {
+                found2 = true;
+                break;
+            }
+        }
+        if (!found2) {
+            /* not in common -> new element */
+            (*new)[*size_new] = strdup(group2[j]);
+            (*size_new)++;
+        }
+    }
+}
+
+/**
+ * @brief Change root dir if enviroment variable "ECONFTOOL_ROOT" exists
+ *
+ * @param path The path to be changed
+ */
+static void changeRootDir(char *path)
+{
+    if (getenv("ECONFTOOL_ROOT") != NULL) {
+        if (strlen(path) + strlen(getenv("ECONFTOOL_ROOT")) >= PATH_MAX) {
+            fprintf(stderr, "ECONFTOOL_ROOT + path is too long\n");
+            exit(EXIT_FAILURE);
+        }
+
+        char *tmp = strdup(path);
+
+        strcpy(path, getenv("ECONFTOOL_ROOT"));
+        strcat(path, tmp);
+
+        free(tmp);
+    }
+}
+
+static int nftwRemove(const char *path, const struct stat *sb, int flag, struct FTW *ftwbuf)
+{
+    return remove(path);
+}
```

Comment:
The docstring for this function is missing.

Reply:
Well, I thought the function plus the name is pretty self-explanatory.

Reply:
Indeed, it is. But for Doxygen this short description is neccessary. I don't know if the project will use Doxygen for documentation, but I started using it when writing econftool.